src/druglab/db/indexing.py

```
@@ -0,0 +1,420 @@
+
+    # Resolve negatives
+    neg_mask = arr < 0
+    if neg_mask.any():
+        arr = arr.copy()
+        arr[neg_mask] = n + arr[neg_mask]
+
+    # Bounds check
+    if n == 0:
+        if len(arr) > 0:
+            raise IndexError(
+                f"Cannot index into an empty dimension (size 0)."
```

Ruff (0.15.10)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)